src/System.EventLog/src/System/Diagnostics/EventLog.cs

```diff
@@ -0,0 +1,1105 @@
+            try
+            {
+                SharedUtils.EnterMutex(eventLogMutexName, ref mutex);
+
+                try
+                {
+                    eventlogkey = GetEventLogRegKey(machineName, true);
+                    if (eventlogkey == null)
+                    {
+                        throw new InvalidOperationException(SR.Format(SR.RegKeyNoAccess, "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Services\\EventLog", machineName));
+                    }
+
+                    using (RegistryKey logKey = eventlogkey.OpenSubKey(logName))
+                    {
+                        if (logKey == null)
+                            throw new InvalidOperationException(SR.Format(SR.MissingLog, logName, machineName));
+
+                        //clear out log before trying to delete it
+                        //that way, if we can't delete the log file, no entries will persist because it has been cleared
+                        EventLog logToClear = new EventLog(logName, machineName);
+                        try
+                        {
+                            logToClear.Clear();
+                        }
+                        finally
+                        {
+                            logToClear.Close();
+                        }
+
+                        string filename = null;
+                        try
+                        {
+                            //most of the time, the "File" key does not exist, but we'll still give it a whirl
+                            filename = (string)logKey.GetValue("File");
+                        }
+                        catch { }
+                        if (filename != null)
+                        {
+                            try
+                            {
+                                File.Delete(filename);
+                            }
+                            catch { }
+                        }
+                    }
+
+                    // now delete the registry entry
+                    eventlogkey.DeleteSubKeyTree(logName);
+                }
+                finally
+                {
+                    eventlogkey?.Close();
+                }
+            }
+            finally
+            {
+                mutex?.ReleaseMutex();
+            }
+        }
+
+        public static void DeleteEventSource(string source)
+        {
+            DeleteEventSource(source, ".");
+        }
+
+        public static void DeleteEventSource(string source, string machineName)
+        {
+            if (!SyntaxCheck.CheckMachineName(machineName))
+            {
+                throw new ArgumentException(SR.Format(SR.InvalidParameter, nameof(machineName), machineName));
+            }
+
+
+            //Check environment before looking at the registry
+            SharedUtils.CheckEnvironment();
+
+            Mutex mutex = null;
+            RuntimeHelpers.PrepareConstrainedRegions();
+            try
```

Comment:
remove all `RuntimeHelpers.PrepareConstrainedRegions();` as we don't support this feature in .NET Core

Reply:
Removed